src/Swarm/TUI/Controller.hs

```diff
@@ -298,13 +298,15 @@ updateUI = do
   let focusedRobotHash = view inventoryHash <$> fr
   -- The hash of the focused robot (if any)
 
+  shouldUpdate <- use (uiState . uiInventoryShouldUpdate)
   -- If the hashes don't match (either because which robot (or
   -- whether any robot) is focused changed, or the focused robot's
   -- inventory changed), regenerate the list.
   inventoryUpdated <-
-    if listRobotHash /= focusedRobotHash
+    if listRobotHash /= focusedRobotHash || shouldUpdate
```

Comment:
I wonder, would it not be enough to check something like `showZero /= listShowsZero`?
I am basing that on `listRobotHash /= focusedRobotHash`, but I have not looked into the code. :slightly_smiling_face: 

If there is some good reason, you can mention it in the `uiInventoryShouldUpdate` docstring :wink:

Reply:
There isn't anywhere that we currently remember something like `listShowsZero`.  I suppose we could add that as another field of the `uiInventory`, but the `uiInventory` is already complicated enough.  I can imagine that in the future we might add other features that would require the inventory list to be updated, in which case we can reuse `uiInventoryShouldUpdate`, whereas something like `showZero /= listShowsZero` is specific to just this toggle.